org.adempiere.base/src/org/compiere/model/MSequence.java

```diff
@@ -488,14 +520,14 @@ public static String getDocumentNoFromSeq(MSequence seq, String trxName, PO po)
 			}
 			else
 			{ // did not find sequence no
-				if (isUseOrgLevel || isStartNewYear) 
+				if (seq.isSequenceNoLevel()) 
 				{	// create sequence (CurrentNo = StartNo + IncrementNo) for this year/month/org and return first number (=StartNo)
 					next = startNo;
 
 					X_AD_Sequence_No seqno = new X_AD_Sequence_No(Env.getCtx(), 0, trxName);
 					seqno.setAD_Sequence_ID(AD_Sequence_ID);
 					seqno.setAD_Org_ID(docOrg_ID);
-					seqno.setCalendarYearMonth(calendarYearMonth);
+					seqno.setSequenceKey(keyParts.getKey());
 					seqno.setCurrentNext(startNo + incrementNo);
 					seqno.saveEx();
```

Comment:
This give the possibility of duplicate AD_Sequence_No record. Perhaps we should have a unique constraint for AD_Sequence_No(AD_Sequence_D, AD_Org_ID, SequenceKey) to prevent this from happening.

Reply:
created the unique constraint